tests/framework/population/test_manager.py

```diff
@@ -489,3 +489,54 @@ class SameColumnCreator(ColumnCreator):
         match="Component 'same_column_creator' is attempting to register private column 'test_column_1' but it is already registered by component 'column_creator'.",
     ):
         InteractiveContext(components=[ColumnCreator(), SameColumnCreator()])
+
+
+def test_extract_columns_from_query() -> None:
+    manager = PopulationManager()
+    query = (
+        # Basic
+        "alive == 'Alive' and is_aged_out == False "
+        # No spaces
+        "age==42 or "
```

Comment:
I'm not sure this case is actually being handled properly, since `"age"` appears in the line below.